### PR TITLE
feat: Omise charge route and payment flow

### LINKED ISSUE #5823: Implement  Omise Payment Gateway
For the OpenTechSummit in Thailand we need local payment gateways. The most popular one is omise. Please implement it.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
omise-final

Frontend: https://github.com/fossasia/open-event-frontend/pull/3073**
Fixes #5823 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
 This creates a charge route for Omise which handles the details from FE, creates a charge object and authorises the payment. This gateway does not require redirection, therefore, can easily be integrated with the android app.

#### Changes proposed in this pull request:

- Create charge route
- Handled all possible exceptions and errors with logging

### CODE REVIEW

File: app/api/orders.py [hunk 1]
```diff
@@ -499,3 +500,27 @@ def alipay_return_uri(order_identifier):
...
+    """
+    Charging the user and returning payment response for Omise Gateway
+    :param order_identifier:
+    :return: JSON response of the payment status.
+    """
+    token = request.form.get("omiseToken")
+    try:
+        charge = OmisePaymentsManager.charge_payment(order_identifier, token)
+    except omise.errors.BaseError as e:
+        logging.error(f"""OmiseError: {repr(e)}.  See https://www.omise.co/api-errors""")
```
Comment (uds5501) [inline app/api/orders.py]: please make the syntax python 3.6 compatible

Comment (iamareebjamal) [reply app/api/orders.py]: Why?

Comment (iamareebjamal) [reply app/api/orders.py]: Upgrade your python please

File: app/api/helpers/payment.py [hunk 2]
```diff
@@ -243,3 +244,26 @@ def charge_source(order_identifier):
...
+
+class OmisePaymentsManager(object):
+    """
+    Class to manage Omise Payments
+    """
+
+    @staticmethod
+    def charge_payment(order_identifier, token):
+        if get_settings()['app_environment'] == Environment.PRODUCTION:
+            omise.api_secret = get_settings()['omise_live_secret']
```
Comment (uds5501) [inline app/api/helpers/payment.py]: Please add `omise.api_public` accordingly

File: app/api/helpers/payment.py [hunk 3]
```diff
@@ -243,3 +244,26 @@ def charge_source(order_identifier):
...
+        else:
+            omise.api_secret = get_settings()['omise_test_secret']
+        order = safe_query(db, Order, 'identifier', order_identifier, 'identifier')
+        charge = omise.Charge.create(
+                amount=order.amount,
```
Comment (uds5501) [inline app/api/helpers/payment.py]: Omise docs needs the amount to be in integer so we should make it
`amount=int(order.amount)`

Comment (mrsaicharan1) [reply app/api/helpers/payment.py]: Yeah, sure

File: app/api/helpers/payment.py (hunk 2 above)
Comment (mrsaicharan1) [reply app/api/helpers/payment.py]: Yup

**Comment by iamareebjamal** [comment]
> Omise docs needs the amount to be in integer so we should make it
amount=int(order.amount)

Umm, so if the order amount is 4.99 dollars, I only charge 4 dollars?

Makes no sense actually

**Comment by mrsaicharan1** [comment]
@iamareebjamal Please check.

**Comment by iamareebjamal** [comment]
Peer Review First

**Comment by mrsaicharan1** [comment]
@uds5501 @shreyanshdwivedi Please review this.

**Comment by mrsaicharan1** [comment]
> Looks good. just take down hound errors please

There are no hound errors! :D

**Comment by mrsaicharan1** [comment]
@iamareebjamal Please merge.

**Comment by mrsaicharan1** [comment]
@iamareebjamal Please merge this.

**Pull request merged**